commands/login.go

```diff
@@ -41,279 +34,34 @@ The password has to be entered interactively or given as -p / --password flag.
 
 The -e or --endpoint argument can be omitted if an endpoint is already selected.`,
 		Example: "  gsctl login user@example.com --endpoint api.example.com",
-		PreRun:  loginPreRunOutput,
-		Run:     loginRunOutput,
+		Run:     loginPickMethod,
 	}
 )
 
-type loginResult struct {
-	// apiEndpoint is the API endpoint the user has been logged in to
-	apiEndpoint string
-	// alias is the alternative, user friendly name for an endpoint
-	alias string
-	// loggedOutBefore is true if the user has been logged out from a previous session
-	loggedOutBefore bool
-	// endpointSwitched is true when the endpoint has been changed during login
-	endpointSwitched bool
-	// email is the email address we are signed in with
-	email string
-	// token is the new session token received
-	token string
-	// numEndpointsBefore is the number of endpoints before login
-	numEndpointsBefore int
-	// numEndpointsAfter is the number of endpoints after login
-	numEndpointsAfter int
-}
-
-type loginArguments struct {
-	apiEndpoint string
-	email       string
-	password    string
-	verbose     bool
-}
-
-func defaultLoginArguments() loginArguments {
-	return loginArguments{
-		apiEndpoint: config.Config.ChooseEndpoint(cmdAPIEndpoint),
-		email:       cmdEmail,
-		password:    cmdPassword,
-		verbose:     cmdVerbose,
-	}
-}
-
 func init() {
 	LoginCommand.Flags().StringVarP(&cmdPassword, "password", "p", "", "Password. If not given, will be prompted interactively.")
+	LoginCommand.Flags().BoolVarP(&cmdSSO, "sso", "", false, "Authenticate using Single Sign On through our identity provider.")
```

Comment:
We wanted this hidden, so we also need

    LoginCommand.Flags().MarkHidden("sso")

here

Reply:
Ok done!